PF/Assignment4/NQueen.java

```diff
@@ -0,0 +1,51 @@
+public class NQueen {
+
+	static boolean solve(int row, int[][] board, int N) {
+		if (row >= N) {
+			return true;
+		}
+		for (int position = 0; position < N; position++) {
+			if (isValid(board, row, position, N)) {
+				board[row][position] = 1;
+				if (!solve(row + 1, board, N)) {
+					board[row][position] = 0;
+				} else
+					return true;
+			}
+		}
+		return false;
+	}
+
+	static int[][] nQueen(int row, int[][] board, int N) {
```

Comment:
ACTION: Since you are using only the constructor (from the test methods), it should be public and rest of the methods (which are not to be used outside) should be compulsorily private.